### PR TITLE
[M] Changed Maven plugin to maven-publish for Gradle 7+ (ENT-4797)

### PR DESCRIPTION
**Pull request by JoshAlbrecht23**
Changed build.gradle to use the maven-publish plugin instead of the maven plugin for compatibility using Gradle 7+.  Unfortunately we are somewhat limited with the properties and methods with maven-publish and I had to generate most of the pom.xml with the withXml method.

https://docs.gradle.org/current/dsl/org.gradle.api.publish.maven.MavenPom.html

**Pom is generated running the following:**
`gradlew generatePomFileForLocalPublication`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -488,285 +488,286 @@ tasks.withType(Checkstyle) {
     exclude "**/i18n/Messages*"
 }
 
-task pom {
```
Comment (Januson) [inline build.gradle:491]: Can you add "pom" alias to the generatePomFileForLocalPublication task?

File: build.gradle [hunk 2]
```diff
@@ -488,285 +488,286 @@ tasks.withType(Checkstyle) {
...
+                createProfileNode(profilesNode, 'build-with-jss4', '/usr/lib64/jss/jss4.jar', '4.9.1')
+                createProfileNode(profilesNode, 'build-with-jss5-plus', '/usr/lib64/jss/jss.jar', '5.0.0')
             }
-        }.writeTo("pom.xml")
+        }
     }
+  }
+}
+
+static void createDependencyNode(Node parentNode, String groupId, String artifactId, String version, String scope, String systemPath) {
```
Comment (Januson) [inline build.gradle:594]: This task grew quite a lot. It should probably move to buildSrc but it depends on how long we will keep it around before migrating away from pom files.

**Review by Januson — CHANGES_REQUESTED** [review]
Changes look good. I have just a small request and a note whether it's worth to move the task out of the buil.gradle file.

File: build.gradle (hunk 1 above)
Comment (JoshAlbrecht23) [reply build.gradle:491]: Good call! I added the "pom" alias

File: build.gradle (hunk 2 above)
Comment (JoshAlbrecht23) [reply build.gradle:597]: I'm not sure if there are plans to move away from the pom files and if it makes sense to move to the buildSrc directory or not. I'm curious on what @Ceiu thinks.

Comment (Ceiu) [reply build.gradle:597]: As far as I know, the pom files are necessary due to the how our build system works and the various pieces within it. We've recently discussed moving to different build systems which wouldn't need the pom files, but I don't think we're very far along on that yet.

As far as compartmentalizing the task, if there's a clean way to do it such that it ends up in a logical place and is self-contained (i.e. not split across several files/modules), that might be better than throwing piles of stuff in the build.gradle file.

Comment (JoshAlbrecht23) [reply build.gradle]: I created a new class, PomToolkit, and moved over the pom xml creation logic to static utility methods that can be used!

**Review by Januson — APPROVED** [review]

**Pull request merged**